### PR TITLE
div. Bugfixes

### PR DESCRIPTION
**Pull request by Moetez16**

### CODE REVIEW

File: app/grails-app/controllers/de/laser/ajax/AjaxController.groovy [hunk 1]
```diff
@@ -1025,6 +1025,7 @@ class AjaxController {
     }
     else {
       log.error("Form submitted with missing values")
+        redirect(url: request.getHeader('referer'))
```
Comment (croqueGrec09) [inline app/grails-app/controllers/de/laser/ajax/AjaxController.groovy:1028]: In an Ajax call?

File: app/grails-app/controllers/de/laser/ajax/AjaxController.groovy [hunk 2]
```diff
@@ -1150,6 +1151,7 @@ class AjaxController {
       }
       else  {
         log.error("Form submitted with missing values")
+          redirect(url: request.getHeader('referer'))
```
Comment (croqueGrec09) [inline app/grails-app/controllers/de/laser/ajax/AjaxController.groovy:1154]: in an ajax call?

File: app/grails-app/services/de/laser/CopyElementsService.groovy [hunk 3]
```diff
@@ -1145,6 +1148,36 @@ class CopyElementsService {
                         newOrgAccessPointLink.subPkg = newSubscriptionPackage
                         newOrgAccessPointLink.save()
                     }
+
+                    subscriptionPackage.getIssueEntitlementsofPackage().each { ie ->
+                        if (ie.status != RDStore.TIPP_STATUS_DELETED) {
+                            def list = subscriptionService.getIssueEntitlements(targetObject).findAll { it.tipp.id == ie.tipp.id && it.status != RDStore.TIPP_STATUS_DELETED }
+                            if (list?.size() > 0) {
```
Comment (croqueGrec09) [inline app/grails-app/services/de/laser/CopyElementsService.groovy:1155]: what did we talk about in today's standup?

File: app/grails-app/views/templates/copyElements/_copyPackagesAndIEs.gsp [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+                                <g:link controller="packageDetails" action="show" target="_blank"
+                                        id="${sp.pkg.id}">${sp.pkg.name}</g:link>
+                                <semui:debugInfo>PkgId: ${sp.pkg.id}</semui:debugInfo>
+                                <g:if test="${sp.pkg.contentProvider}">(${sp.pkg.contentProvider.name})</g:if>
+                                <br>
+                                <g:set var="ies" value="${sp.getIssueEntitlementsofPackage()}"/>
+
+                                <div class="ui accordion">
+                                    <div class="title"><i
+                                            class="dropdown icon"></i> ${message(code: 'issueEntitlement.countSubscription')} </strong>${ies ? ies.size() : ""}
```
Comment (croqueGrec09) [inline app/grails-app/views/templates/copyElements/_copyPackagesAndIEs.gsp:162]: can ies be null?

File: app/grails-app/views/templates/copyElements/_copyPackagesAndIEs.gsp [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+                    </div>
+                </th>
+            </tr>
+            </thead>
+            <tbody class="top aligned">
+            <tr>
+                <g:set var="excludes"
+                       value="${[PendingChangeConfiguration.PACKAGE_PROP, PendingChangeConfiguration.PACKAGE_DELETED]}"/>
+                <td name="subscription.takePackages.source">
+                    <strong>${message(code: 'subscription.packages.label')}: ${sourceObject.packages?.size()}</strong>
```
Comment (croqueGrec09) [inline app/grails-app/views/templates/copyElements/_copyPackagesAndIEs.gsp:56]: packages?.size()

File: app/grails-app/views/templates/copyElements/_copyPackagesAndIEs.gsp [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+                                <br/>
+                            </div>
+                        </div>
+                    </g:each>
+                </td>
+
+
+
+                <td name="subscription.takePackages.target">
+                    <strong>${message(code: 'subscription.packages.label')}: ${targetObject?.packages?.size()}</strong>
```
Comment (croqueGrec09) [inline app/grails-app/views/templates/copyElements/_copyPackagesAndIEs.gsp:147]: packages?.size()

File: app/grails-app/services/de/laser/CopyElementsService.groovy (hunk 3 above)
Comment (Moetez16) [reply app/grails-app/services/de/laser/CopyElementsService.groovy:1155]: Copy & Pasta. Vergessen ;(

**Review by croqueGrec09 — COMMENTED** [review]
unnecessary ?

**Review by klodav — APPROVED** [review]

**Review by croqueGrec09 — APPROVED** [review]
approved

**Pull request merged**